src/jdk.jdwp.agent/share/native/libjdwp/classTrack.c

```diff
@@ -78,14 +93,20 @@ addPreparedClass(JNIEnv *env, jclass klass)
 
     char* signature;
     error = classSignature(klass, &signature, NULL);
+    if (is_wrong_phase(error)) {
+        return;
+    }
     if (error != JVMTI_ERROR_NONE) {
         EXIT_ERROR(error,"signature");
     }
 
-    if (gdata && gdata->assertOn) {
+    if (gdata->assertOn) {
```

Comment:
Just a question.
Why the check for non-NULL gdata was removed?
Is it because gdate is always non-NULL?

Reply:
It's created during the OnLoad and never freed, so it can never be null once this callback is setup.